pintc/tests/storage/new_type_map.pnt

```diff
@@ -0,0 +1,31 @@
+type MyMap = (int => int);
+type MyInt = int;
+type MyNestMap = (MyInt => int);
+
+storage {
+    m: MyMap,
+    n: MyInt,
+    o: MyNestMap,
+}
+
+// parsed <<<
+// storage {
+//     m: ::MyMap,
+//     n: ::MyInt,
+//     o: ::MyNestMap,
+// }
+// type ::MyMap = ( int => int );
+// type ::MyInt = int;
+// type ::MyNestMap = ( ::MyInt => int );
+// >>>
+
+// flattened <<<
+// storage {
+//     m: ( int => int ),
+//     n: int,
+//     o: ( ::MyInt (int) => int ),
```

Comment:
This doesn't seem to have lowered the inner alias.  This should happen in `transform::lower_aliases()` in `lower.rs`.

When glancing at the code in there I see it has a recursive inner function for finding alias types and replacing them with the aliased type, but it doesn't go further when the replace type is also an alias.

So here it's replacing `MyNestMap` with `(MyInt => int)` but it doesn't bother continuing to replace `MyInt` with `int`.  I have a feeling this is a problem in general, not just this PR.  As in an alias of an alias probably won't be lowered properly for `var` or `state` types either.

The solution I think is to iterate until there are no changes/replacements made, in the same fashion as other transformation passes, since the replacing type could itself be quite complex and have many nested type aliases.

And so this is a tricky enough problem to warrant its own issue.  I'll do that now and approve this PR.